cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -223,7 +225,7 @@ func getClusterFilter(kubeClient *kubernetes.Clientset, settingsMgr *settings.Se
 		appControllerDeployment = nil
 	}
 
-	if appControllerDeployment != nil && appControllerDeployment.Spec.Replicas != nil {
+	if appControllerDeployment != nil && appControllerDeployment.Spec.Replicas != nil && dynamicClusterDistributionEnabled {
```

Comment:
Let's move this to be the first check. 

Also, to expand further, we don't need to check for Deployments if we are introducing this flag.

Reply:
We still need to check for deployments to avoid a nil pointer dereference. Updated to check for the flag first